### PR TITLE
Misc small updates

### PR DESCRIPTION
**Pull request by netsettler**
Various updates:

* Extend the documentation of the big change for `dcicutils 4.0.0` in `CHANGELOG.rst`.
* Add unit testing for stray `print(...)` and `pdb.set_trace`.
* Support for `ENCODED_CREATE_MAPPING_SKIP`, `ENCODED_CREATE_MAPPING_WIPE_ES`, and `ENCODED_CREATE_MAPPING_STRICT` in GAC to allow `$CREATE_MAPPING_SKIP`, `$CREATE_MAPPING_WIPE_ES`, and `$CREATE_MAPPING_STRICT` in `.ini` files.
* Allow `get_foursight_bucket` to infer a bucket prefix if one is not explicitly supplied. (It removes `-envs` from the global env bucket name and uses that.)
* Fix various tests that had grown stale due to data changes. See CHANGELOG for details.

### CODE REVIEW

File: test/test_misc.py [hunk 1]
```diff
@@ -78,3 +80,10 @@ def test_documentation():
             message = there_are(problems, kind="problem", tense='past', show=False,
                                 context=f"found in the readthedocs declaration file, {_DCICUTILS_DOC_FILE!r}")
             raise AssertionError(message)
+
+
+def test_for_debugging_print():
+    confirm_no_uses(where=os.path.join(LIBRARY_DIR, "*.py"), patterns={
+        "call to print": "^[^#]*print[(]",
+        "active use of pdb.set_trace": "^[^#]*pdb[.]set_trace[(][)]"
+    })
```
Comment (dmichaels-harvard) [inline test/test_misc.py:89]: Might want a test to make sure an exception is raise if a print actually exists, e.g. I tried this:
`    import pytest
    import tempfile
    with tempfile.NamedTemporaryFile(dir=LIBRARY_DIR, suffix=".py", mode="w") as f:
        f.write("print('helloworld')")
        f.flush()
        with pytest.raises(RuntimeError):
            confirm_no_uses(where=os.path.join(LIBRARY_DIR, "*.py"), patterns={
                "call to print": "^[^#]*print[(]",
                "active use of pdb.set_trace": "^[^#]*pdb[.]set_trace[(][)]"
            })
    with tempfile.NamedTemporaryFile(dir=LIBRARY_DIR, suffix=".py", mode="w") as f:
        f.write("import pdb ; pdb.set_trace() ")
        f.flush()
        with pytest.raises(RuntimeError):
            confirm_no_uses(where=os.path.join(LIBRARY_DIR, "*.py"), patterns={
                "call to print": "^[^#]*print[(]",
                "active use of pdb.set_trace": "^[^#]*pdb[.]set_trace[(][)]"
            })
`

Comment (netsettler) [reply test/test_misc.py:89]: I actually used the tool to eliminate a bunch of cases of both, but a test is a good idea, of course. Thanks for prototyping things. I'll see about putting something like that in.

**Review by dmichaels-harvard — APPROVED** [review]

File: test/test_misc.py (hunk 1 above)
Comment (netsettler) [reply test/test_misc.py]: I added some tests.

**Review by willronchetti — APPROVED** [review]

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,25 +6,62 @@ dcicutils
...
+  the abstraction level of ``authorized_request``.
+* Fix various tests that had grown stale due to data changes.
+  * ``test_post_delete_purge_links_metadata`` (needed to be re-recorded)
+  * ``test_upsert_metadata`` (needed to be re-recorded)
+  * ``test_unified_authentication_prod_envs_integrated_only`` 
+    (simplified, removed bogus attempts at recording)
+  * ``test_faceted_search_exp_set`` (needed many different counts)
+  * ``test_some_decorated_methods_work`` (needed one different count)
+  * ``test_faceted_search_exp_set`` (newly recorded)
+  * ``test_faceted_search_users`` (newly recorded)
```
[the remark this answers is not in the document]
Comment (netsettler) [reply CHANGELOG.rst:30]: Yeah, I did that already. Tests seem to be passing everywhere.

**Pull request merged**